src/ring_theory/int/basic.lean

```diff
@@ -276,7 +276,7 @@ begin
   cases n, { simp },
   rw [← multiset.rel_eq, ← associated_eq_eq],
   apply factors_unique (irreducible_of_normalized_factor) _,
-  { rw [multiset.coe_prod, nat.prod_factors (nat.succ_pos _)],
+  { rw [multiset.coe_prod, nat.prod_factors n.succ_ne_zero],
```

Comment:
This PR is making lots of changes like this. It would be good to split these off into another PR. Because it is currently touching 10 files, and the `factorization` stuff is enough of a change that the PR shouldn't be bigger than needed.

Reply:
I agree with you in principle, but I think in this case it's particularly hard to decouple these changes, as the proof edits are not just cosmetic but due to the (slight) generalisation in lemmas. Also, since I have a lot of stuff depending on this PR in unit-fractions, do you think it's okay if we keep this as one PR for now - especially because this refactor is quite likely to break if there are any new lemmas related to factorization, and in the future I'll make a better effort to split up PRs more?